kratos/mpi/utilities/parallel_fill_communicator.cpp

```diff
@@ -194,19 +194,36 @@ void ParallelFillCommunicator::PrintData(std::ostream& rOStream) const
 
 void ParallelFillCommunicator::ComputeCommunicationPlan(ModelPart& rModelPart)
 {
+    KRATOS_TRY;
+
+    KRATOS_ERROR_IF_NOT(rModelPart.HasNodalSolutionStepVariable(PARTITION_INDEX)) << "\"PARTITION_INDEX\" missing as solution step variable in ModelPart \"" << rModelPart.Name() << "\"!" << std::endl;
+
     constexpr unsigned root_id = 0;
 
     Communicator::Pointer pnew_comm = Kratos::make_shared< MPICommunicator >(&rModelPart.GetNodalSolutionStepVariablesList(), DataCommunicator::GetDefault());
     rModelPart.SetCommunicator(pnew_comm);
 
     const auto& r_data_communicator = pnew_comm->GetDataCommunicator();
 
+    // Check if the nodes have been assigned a partition index (i.e. some value different from 0). If not issue a warning
+    int non_zero_partition_index_found = 0;
+    for (const auto& r_node : rModelPart.Nodes()) {
+        const int node_partition_index = r_node.FastGetSolutionStepValue(PARTITION_INDEX);
+        if (node_partition_index != 0) {
+            non_zero_partition_index_found = 1;
+            break;
+        }
+    }
+
+    pnew_comm->SumAll(non_zero_partition_index_found);
```

Comment:
Note that DataCommunicator should have an OrAll option for bool (and, if it is not there yet, I probably should add it)

Reply:
I suspected that there would be a better solution, but I didn't find it, could you point to what you would do?

Reply:
The `DataCommunicator::OrReduceAll` method would fit nicely here, unfortunately I just realized that it is part of #5929. I can try separating it into a smaller PR, but I would not block this PR for that.